app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,30 @@ def show
     @post = Post.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: post } }
+    end
+  end
+
+  def create
+    post = Post.new(params.require(:post).permit(:title, :text))
+    user = User.find(params[:user_id])
+    post.author = user
+    post.comments_counter = 0
+    post.likes_counter = 0
```

Comment:
- [ ] <b>REQUIRED:</b> I don't really get the reason you trying to reassign those values since they are already initialized. post is coming from the database so, all of those properties must have values. if they don't then you must have a mistake in the previous milestone.